static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -12,6 +12,17 @@ import type RequestError from 'sentry/utils/requestError/requestError';
 import useProjects from 'sentry/utils/useProjects';
 import type {ReplayError, ReplayRecord} from 'sentry/views/replays/types';
 
+// Subtract & add a few seconds from `started_at` and `finished_at` to account
+// for errors that might've come in with some clock drift, and are therefore
+// outside the the start/end times when we got segment data.
+
+// We need to add at least 1ms to `finished_at` date because
+// finished_at has the `ms` portion truncated, while replays-events-meta operates on
+// timestamps with `ms` attached. So finishedAt could be at time `12:00:00.000Z`
+// while the event is saved with `12:00:00.450Z`.
+const START_TIME_FUZZ_MS = -2_000;
```

Comment:
Is 2s decided arbitrarily?

Reply:
yup. i found out later the the clock-drift problem could be as severe as a few hours, so things would need to change to accomodate that magnitude.